src/check/arbitrary/ArrayArbitrary.ts

```diff
@@ -90,7 +90,13 @@ export class ArrayArbitrary<T> extends Arbitrary<T[]> {
     if (items.length === 0) {
       return Stream.nil<Shrinkable<T>[]>();
     }
-    const size = this.lengthArb.shrinkableFor(items.length, shrunkOnce);
+    const size = this.lengthArb.contextualShrinkableFor(
```

Comment:
This change has zero shrinker impacts. In order to benefit from this change we should re-use the instances generated during the shrinking process of size. But It would imply a deeper refactoring of array arbitrary.